dask/dataframe/core.py

```diff
@@ -1855,21 +1855,43 @@ def astype(self, dtype):
         empty = self._pd.astype(dtype)
         return map_partitions(pd.DataFrame.astype, empty, self, dtype=dtype)
 
-    def info(self):
+    def info(self, buf=None, verbose=False, memory_usage=False):
         """
         Concise summary of a Dask DataFrame.
         """
-        lines = list()
-        lines.append(str(type(self)))
-        lines.append('Data columns (total %d columns):' % len(self.columns))
-        dtypes = self.dtypes
-        space = max([len(k) for k in self.columns]) + 4
-        template = "%s%s"
-        for i, col in enumerate(self.columns):
-            dtype = dtypes.iloc[i]
-            lines.append(template % (('%s' % col)[:space].ljust(space), dtype))
-
-        print('\n'.join(lines))
+
+        lines = [str(type(self))]
+
+        if buf is None:
+            import sys
+            buf = sys.stdout
+
+        if len(self.columns) == 0:
+            lines.append('Index: 0 entries')
+            lines.append('Empty %s' % type(self).__name__)
+            put_lines(buf, lines)
+            return
+
+        column_template = "{0:<%d} {1}" % (self.columns.str.len().max() + 5)
+
+        if verbose:
+            index, counts = da.compute(self.index, self.count())
+            lines.append(index.summary())
+            column_template = column_template.format('{0}', '{1} non-null {2}')
+            column_info = [column_template.format(*x) for x in zip(self.columns, counts, self.dtypes)]
+        else:
+            column_info = [column_template.format(*x) for x in zip(self.columns, self.dtypes)]
+
+        lines.append('Data columns (total {} columns):'.format(len(self.columns)))
+        lines.extend(column_info)
+        dtype_counts = ['%s(%d)' % k for k in sorted(self.dtypes.value_counts().iteritems())]
+        lines.append('dtypes: {}'.format(', '.join(dtype_counts)))
+
+        if memory_usage:
+            memory_int = self.map_partitions(pd.DataFrame.memory_usage, index=True).compute().sum()
+            lines.append('memory usage: {}\n'.format(memory_repr(memory_int)))
```

Comment:
There are two separate calls to compute here.  In the single-machine on-disk case this will require two reads through data rather than one.  This isn't a huge deal because it's within an optional case but still it might be nice to bundle all computations into a single call to compute.  This gets a bit awkward of course with the different options (which is probably why you opted for two separate calls to compute).